### ISSUE TITLE
"Ao Channel Affect" defaults to 0, giving unexpected results (fixed in `master` via different AO calculation)

### ISSUE THREAD

**Issue by Arnklit** [godotengine/godot#43957]
**Godot version:**
3.2.3.Stable

**OS/device including version:**
NA

**Issue description:**
"Ao Channel Affect" defaults to 0, giving unexpected results. This means that as soon as you set an AO texture on a material, there will no longer be SSAO on that material. When this happened to me I assumed it was simply a limitation of the engine and on purpose. I think we should default that setting to 1. From talking to other Godot users I know others have thought that you couldn't use SSAO and AO textures at the same time because of this.

I'm also wondering if "Ao Channel Affect" is actually doing the correct thing. Currently it's controlling how much the SSAO can influence the AO maps, but it feels more logical that it would be how much the the AO maps are allowed to influence the SSAO.

![image](https://user-images.githubusercontent.com/4955051/100543411-e0c4e900-3247-11eb-863c-e9bab6f7bd8a.png)

**Steps to reproduce:**
Open the attached project to the see issue.

**Minimal reproduction project:**
[ssao_aompap_issue.zip](https://github.com/godotengine/godot/files/5612196/ssao_aompap_issue.zip)

**Comment by Calinou** [comment]
I think the AO textures were initially designed to be used on 3D meshes where you don't want SSAO to be applied. Otherwise, the occlusion will be twice as strong as originally intended.

For "world" materials, you should definitely set **Ao Channel Affect** to 1.

**Comment by Arnklit** [comment]
Yeah I sort of figured that was the idea. The example of an AO texture in the manual is also for larger shapes with an AO texture applied, but that's not really how AO textures are used in modern game engines, they are baked out to add the small details that cannot be captured by SSAO.

But I don't think it should be a problem that the occlusion get's "twice as strong". if the blend mode is correct it should take whatever was darkest between the SSAO and texture I assume?

**Comment by Calinou** [comment]
This was fixed by https://github.com/godotengine/godot/pull/45023 in the `master` branch.

While this is technically a breaking change, I wonder if we should change the default value in `3.x` consider the default behavior is quite broken – or backport the AO change from `master` to `3.x` instead. This would be for 3.4 or later, not 3.3.x.

**Comment by Arnklit** [comment]
I'm happy to close this as it's resolved in 4.x and I think the people still on 3.x are probably used to the old behavior and unlikely to expect changes in this.

**Comment by Calinou** [comment]
Closing per @Arnklit's comment.
